File: electron/db/json.ts
/**
 * SQLite JSON 数组字段的安全序列化/反序列化工具。
 * 领域对象中数组字段在入库前 JSON.stringify，出库后 JSON.parse。
 */

export function parseJsonArray<T = string>(value: string | null | undefined): T[] {
  if (!value) return []
  try {
    const parsed: unknown = JSON.parse(value)
    return Array.isArray(parsed) ? (parsed as T[]) : []
  } catch {
    return []
  }
}

export function stringifyJsonArray(value: unknown[] | undefined | null): string {
  if (!value) return '[]'
  try {
    return JSON.stringify(value)
  } catch {
    return '[]'
  }
}

/** 安全解析任意 JSON 字段（对象或数组） */
export function parseJsonField<T>(value: string | null | undefined, fallback: T): T {
  if (!value) return fallback
  try {
    return JSON.parse(value) as T
  } catch {
    return fallback
  }
}

File: src-tauri/src/db/json.rs
/** JSON 工具：安全解析/序列化 JSON 数组与字段（对应 electron/db/json.ts） */

use serde::de::DeserializeOwned;
use serde::Serialize;

/// 安全解析 JSON 数组，失败返回空 Vec
pub fn parse_json_array<T: DeserializeOwned>(value: &str) -> Vec<T> {
    if value.is_empty() {
        return Vec::new();
    }
    serde_json::from_str(value).unwrap_or_default()
}

/// 安全序列化为 JSON 数组字符串，失败返回 "[]"
pub fn stringify_json_array<T: Serialize>(value: &[T]) -> String {
    serde_json::to_string(value).unwrap_or_else(|_| "[]".to_string())
}

/// 安全解析任意 JSON 字段，失败返回 fallback
pub fn parse_json_field<T: DeserializeOwned>(value: &str, fallback: T) -> T {
    if value.is_empty() {
        return fallback;
    }
    serde_json::from_str(value).unwrap_or(fallback)
}
